src/Plugin/search_api_autocomplete/suggester/Spellcheck.php

```diff
@@ -15,11 +15,10 @@
  * The server needs to support the "search_api_autocomplete" feature for this to
  * work.
  *
- * @SearchApiAutocompleteSuggester(
- *   id = "search_api_solr_spellcheck",
- *   label = @Translation("Solr Spellcheck"),
- *   description = @Translation("Suggest corrections for the entered words based on Solr's spellcheck component. Note: Be careful when activating this feature if you run multiple indexes in one Solr core! The spellcheck component is not able to distinguish between the different indexes and returns suggestions for the complete core. If you run multiple indexes in one core you might get suggestions that lead to zero results on a specific index!"),
- * )
+ * @deprecated in search_api_solr:4.3.0 and is removed from search_api_solr:5.0.0. Use the
+ *   \Drupal\search_api_solr_autocomplete\Plugin\search_api_autocomplete\suggester\Spellcheck instead
+ *
+ * @see https://www.drupal.org/node/3254186
  */
 class Spellcheck extends SuggesterPluginBase implements PluginFormInterface {
```

Comment:
It's not enough to add the `@deprecated` tag in order to deprecate. Please read again and carefully https://www.drupal.org/about/core/policies/core-change-policies/drupal-deprecation-policy#how-class. We're treating now this as a class as it lacks the plugin annotation.

Also, a deprecated class should only point to the new code and remove their code:

```php

use Drupal\search_api_solr_autocomplete\Plugin\search_api_autocomplete\suggester\Spellcheck as SpellcheckOriginal

class Spellcheck extends SpellcheckOriginal {}
```

Please treat in the same way `Suggester`, `Terms`